TreeDB/docs/evidence/vector-partition-local-system-qualification-4019/m2-95c60cbe/run_m2.py

```
@@ -0,0 +1,213 @@
+    head = subprocess.run(["git", "-C", str(ROOT / "source"), "rev-parse", "HEAD"], check=True, capture_output=True, text=True).stdout.strip()
+    if status or head != SOURCE_HEAD:
+        raise RuntimeError("source checkout changed")
+    for entry in Path("/proc").iterdir():
+        if not entry.name.isdigit():
+            continue
+        try:
+            comm = (entry / "comm").read_text(encoding="utf-8").strip()
+        except (FileNotFoundError, ProcessLookupError, PermissionError):
+            continue
+        if comm in ("go", "KaHIP") or comm.endswith(".test") or comm == "treedb_vector_partition_bench.vcs":
+            raise RuntimeError(f"unexpected live heavy process {entry.name} {comm}")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)